### ISSUE TITLE
How many records are available on the api?

### ISSUE THREAD

**Issue by Wambere** [OpenSRP/opensrp-server-web#549]
When we pull data from the api, in order to confirm that we have pulled all the data that is available, we have been using the following queries:
- select count(*) from core.event_metadata where date_deleted is null;
- select count(*) from core.task_metadata;
- select count(*) from core.plan;
- select count(*) from core.location_metadata where status='ACTIVE';
- select count(*) from core.structure_metadata;
- select count(*) from core.client_metadata where date_deleted is null;

We have however found out that these do not actually match what is available from the api. For example

- `select count(*) from core.event_metadata where date_deleted is null;` returns `412798` while pulling all events from the getAll endpoint returns `410850` events
- `select count(*) from core.task_metadata;` returns `2035185` yet pulling all tasks from the getAll endpoint is currently returning `2034776`
- (This is more weird) `select count(*) from core.location_metadata where status='ACTIVE';` returns `49431`, but when I tried hitting the api with `https://reveal-zm.smartregister.org/opensrp/rest/location/findByJurisdictionIds?jurisdiction_ids=${id}` with all the IDs in the jurisdiction table, `52155` records had a valid, non empty response. And querying `select count(*) from core.location_metadata;` returns `52575`

Basically, it's impossible to manage sync without knowing exactly how many records to expect. 
Is it possible to just add count endpoints and get these also from the api?

**Comment by Wambere** [comment]
Hi @bennsimon , in the above, the limit when making a request is not required, but is set to a default of 25. So I basically need to pass in a limit of a number higher than what is in the database to get a full count, otherwise it always returns 25. Not sure I understand why or how this is useful. 
This would mean we either need to know what the count is to pass a "reasonable" limit and we would need to keep changing this as the numbers grow, or maybe set it to a super high  number which we think we will never reach. 

Is there an alternative option which gets the count without needing to pass in this limit?

**Comment by bennsimon** [comment]
Cool, let me make adjustments to the api.
